arch/rv32i/src/pmp.rs

```diff
@@ -136,37 +152,29 @@ impl PMPRegion {
     }
 }
 
-pub trait PMPConfigType: Default + Copy + Clone + Sized {}
-
 /// Struct storing region configuration for RISCV PMP.
-pub struct PMPConfig<N: PMPConfigType> {
+pub struct PMPConfig {
```

Comment:
This change is fine, but just noting that it is moving us further towards the macro approach as we get further away from the array generators.

Reply:
For general awareness, the minimum subset of const generics that will be stabilized now has its own feature gate (https://github.com/rust-lang/rust/pull/74877). Its not stable yet, but clearly on the fast track now.